packages/platform-sdk/test/integration-tests/sdk-v3/middlewares.test.ts

```diff
@@ -187,6 +189,90 @@ describe('Concurrent Modification Middleware', () => {
   })
 })
 
+describe('Http clients and http client options', () => {
+  it('Call the `httpClient` using the provided client options - [throw]', async () => {
+    let isCalled = false
+
+    const after = () => {
+      return (next: Next) => {
+        return (request: MiddlewareRequest) => {
+          isCalled = true
+          return next(request)
+        }
+      }
+    }
+
+    const http: HttpMiddlewareOptions = {
+      ...httpMiddlewareOptionsV3,
+      httpClient: axios,
+      host: 'https://api.us-central1.gcp.commercetools.com', // should fail
+      httpClientOptions: {
+        validateStatus: () => false,
+      },
+    }
+
+    const v3Client = new ClientBuilderV3()
+      .withClientCredentialsFlow(authMiddlewareOptionsV3)
+      .withHttpMiddleware(http)
+      // should not be called (since axios will throw internal error)
+      .withAfterExecutionMiddleware({ middleware: after })
+      .build()
+
+    const api = createApiBuilderFromCtpClient(v3Client).withProjectKey({
+      projectKey,
+    })
+
+    await api
+      .get()
+      .execute()
+      .catch(() => null)
+    expect(isCalled).toBe(false)
+  })
+
+  it('Call the `httpClient` using the provided client options - [not throw]', async () => {
+    let isCalled = false
+
+    const after = () => {
+      return (next: Next) => {
+        return (request: MiddlewareRequest) => {
+          isCalled = true
+          return next(request)
+        }
+      }
+    }
+
+    const auth = {
+      ...authMiddlewareOptionsV3,
+    }
+
+    const http: HttpMiddlewareOptions = {
+      ...httpMiddlewareOptionsV3,
+      httpClient: axios,
+      host: 'https://api.us-central1.gcp.commercetools.com', // should fail
```

Comment:
Yes, the host is in `us` and the tests credentials are for `eu`, hence should throw a 404.

Reply:
I see, then it's better to add this fact somewhere so it's a precondition for this test. Because right now if the credentials are changed to US, then this test would suddenly work.

Reply:
I used an incorrect generic host, so no need to worry if anything is changed in the future.